swarm_dial.go

```diff
@@ -383,20 +403,28 @@ func (s *Swarm) dial(ctx context.Context, p peer.ID) (*Conn, error) {
 		return nil, &DialError{Peer: p, Cause: ErrNoAddresses}
 	}
 	goodAddrs := s.filterKnownUndialables(p, peerAddrs)
+	// TODO Consider dialing private addresses as well
+	// See https://github.com/libp2p/specs/pull/173/files#r288784559
+	if forceDirect {
+		goodAddrs = addrutil.FilterAddrs(goodAddrs, manet.IsPublicAddr, s.nonProxyAddr)
+	}
```

Comment:
hrm... yeah, this could be problematic here. If we are behind the same NAT we want the private dial. 
Let's not filter only public addrs?